File: solutions/204.count-primes/count-primes.js
/**
 * @param {number} n
 * @return {number}
 */
const countPrimes = n => {
  const isPrime = new Array(n).fill(true)

  let count = 0

  for (let i = 2; i < n; i++) {
    if (isPrime[i]) {
      count++
      for (let j = 0; i * j < n; j++) {
        isPrime[i * j] = false
      }
    }
  }

  return count
}

File: solutions/204.count-primes/count-primes.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        0 0 0 0 0
        0 1 2 3 1
        """
        not_primie = [0]*n
        count = 0
        for i in range(2, n):
            j = 2

            if not_primie[i] == 0:
                count += 1

            while i*j < n:
                not_primie[i*j] = 1
                j += 1

        return count
